### PR TITLE
[Feature Request] Expose all log fields and tags contained in log records

### PR DESCRIPTION
**Pull request by eldios**
As per customer request (customer M.) changed code to expose all log fields and tags contained in log records.

### CODE REVIEW

File: cmd/adapter/api/api.go [hunk 1]
```diff
@@ -357,6 +365,23 @@ func toGravityLogEntries(evs []*api.LogEvent) ([]string, error) {
...
+
+	// iterate over the elements in the form `elem1, elem2, elem3, ...`
+	for _, elem := range strings.Split(csv, ",") {
+
+		// each element should be in the form `key=value`
+		elemSplit := strings.Split(elem, "=")
+		if len(elemSplit) == 2 {
+			results[elemSplit[0]] = elemSplit[1]
+		} else {
+			// one of the element was not in the form `key=value`
```
Comment (knisbet) [inline cmd/adapter/api/api.go:385]: Is it not valid to have just a key as a tag?

**Review by knisbet — APPROVED** [review]

File: cmd/adapter/api/api.go [hunk 2]
```diff
@@ -357,6 +365,23 @@ func toGravityLogEntries(evs []*api.LogEvent) ([]string, error) {
 	return entries, nil
 }
 
+func parseCSVIntoMap(csv string) (results map[string]string) {
+
+	// iterate over the elements in the form `elem1, elem2, elem3, ...`
+	for _, elem := range strings.Split(csv, ",") {
+
+		// each element should be in the form `key=value`
+		elemSplit := strings.Split(elem, "=")
```
Comment (a-palchikov) [inline cmd/adapter/api/api.go:378]: Can you describe the input format in more detail - maybe as a Godoc for the method?
Can a value be a string with whitespace and potentially include an embedded equals sign if quoted?

File: cmd/adapter/api/api.go (hunk 1 above)
Comment (eldios) [reply cmd/adapter/api/api.go:385]: Added case where there's a an empty tag with just the key.

Thanks!

File: cmd/adapter/api/api.go (hunk 2 above)
Comment (eldios) [reply cmd/adapter/api/api.go:378]: explanation comment and FIXME about the '=' char added.

**Pull request merged**